app/playlists.js

```diff
@@ -0,0 +1,50 @@
+/*-----------Playlist Class Implementation----------------------------------*/
+
+function Playlist(id,title,description,img){
+  this.id = id;
+  this.songList = [];
+  this.description = description;
+  this.img = img;
+  this.title = title;
+}
+// Playlist Class Methods
+Playlist.prototype={
+  getTitle:function(){
+    return this.title;
+  },
+  setTitle:function(newTitle){
+    this.title = newTitle;
+  },
+  getId:function(){
+    return this.id;
+  },
+  setId:function(id){
+    this.id=id;
+  },
+  getSongList:function(){
+    return this.songList;
+  },
+  setSongList:function(list){
+    //!!! implement to automatically populate playlist with Song objects
+    for(i=0;i<list.length;i++){
+        this.addSong(new Song(list[i].songTitle, list[i].songAuthor, list[i].image, list[i].songLength));
+    }
+    this.songList=list;
+  },
+  addSong:function(newSong){
+    this.songList.push(newSong);
```

Comment:
Use instanceof to ensure that newSong is of type Song( and not just a plain JS object let's say)